runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CoderTranslator.java

```diff
@@ -38,7 +38,11 @@
   List<? extends Coder<?>> getComponents(T from);
 
   /**
-   * Create a {@link Coder} from its component {@link Coder coders}.
+   * Returns the serialized payload that will be provided when deserializing this coder, if any. If
+   * there is no payload, a byte array of length 0 should be returned.
    */
-  T fromComponents(List<Coder<?>> components);
+  byte[] getPayload(T from);
```

Comment:
would default implementations here make sense instead of adding the `SimpleStructuredCoderTranslator`?

Reply:
Default for this method is good; I don't think there's a useful default implementation of `fromComponents`, so the abstract class is still somewhat useful.